slatedb/src/compactor_state.rs

```diff
@@ -52,44 +51,334 @@ impl SourceId {
     }
 }
 
-#[derive(Clone, Debug, PartialEq)]
-pub(crate) enum CompactionStatus {
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum CompactorJobStatus {
     Submitted,
     #[allow(dead_code)]
+    Pending,
+    #[allow(dead_code)]
     InProgress,
+    #[allow(dead_code)]
+    Completed,
+    #[allow(dead_code)]
+    Failed,
 }
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub(crate) enum CompactorJobRequestType {
+    /// Signals that the compaction was requested by the DB's compactor.
+    Internal,
+    /// Signals that the compaction was requested by an external process such as the admin CLI.
+    #[allow(dead_code)]
+    External,
+}
+
+/// In-memory description of the inputs to a compaction.
+///
+/// A `CompactionSpec` represents the concrete set of input SSTs and/or Sorted Runs
+/// that a particular compaction algorithm (e.g., size-tiered, leveled, etc.) has
+/// selected to produce a destination Sorted Run.
+///
+/// The scheduler constructs a `CompactionSpec` for a planned compaction and passes it,
+/// together with the selected `sources`, to `Compaction::new`. Keeping the spec inside
+/// the in-memory `Compaction` avoids recomputing or cloning inputs at job creation time
+/// and enables encode/decode roundtrips in tests via FlatBuffers.
 #[derive(Clone, Debug, PartialEq)]
-pub struct Compaction {
-    pub(crate) status: CompactionStatus,
-    pub(crate) sources: Vec<SourceId>,
-    pub(crate) destination: u32,
+pub(crate) enum CompactorJobInput {
```

Comment:
I'm wondering if it would make more sense to not use an enum for the in-memory model for this for now. I think for the flatbuffers we want a union to give us the chance to evolve it, but for the in-memory model it might be simpler to just assume we're always doing compaction of whole SRs